pkg/cmd/operator/mustgather_test.go

```diff
@@ -0,0 +1,224 @@
+package operator
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/gather/collect/testhelpers"
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"github.com/spf13/cobra"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	dynamicfakeclient "k8s.io/client-go/dynamic/fake"
+	kubefakeclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/klog/v2"
+)
+
+func TestMustGatherOptions_Run(t *testing.T) {
+	apiResources := []*metav1.APIResourceList{
+		{
+			GroupVersion: corev1.SchemeGroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "namespaces", Namespaced: false, Kind: "Namespace", Verbs: []string{"list"}},
+				{Name: "pods", Namespaced: true, Kind: "Pod", Verbs: []string{"list"}},
+				{Name: "secrets", Namespaced: true, Kind: "Secret", Verbs: []string{"list"}},
+			},
+		},
+		{
+			GroupVersion: admissionregistrationv1.SchemeGroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "validatingwebhookconfigurations", Namespaced: false, Kind: "ValidatingWebhookConfiguration", Verbs: []string{"list"}},
+				{Name: "mutatingwebhookconfigurations", Namespaced: false, Kind: "MutatingWebhookConfiguration", Verbs: []string{"list"}},
+			},
+		},
+		{
+			GroupVersion: scyllav1alpha1.GroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "scyllaoperatorconfigs", Namespaced: true, Kind: "ScyllaOperatorConfigs", Verbs: []string{"list"}},
+				{Name: "nodeconfigs", Namespaced: true, Kind: "NodeConfigs", Verbs: []string{"list"}},
+			},
+		},
+		{
+			GroupVersion: scyllav1.GroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "scyllaclusters", Namespaced: true, Kind: "ScyllaCluster", Verbs: []string{"list"}},
+			},
+		},
+	}
+
+	testScheme := runtime.NewScheme()
+
+	err := corev1.AddToScheme(testScheme)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = admissionregistrationv1.AddToScheme(testScheme)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = scyllav1alpha1.Install(testScheme)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = scyllav1.Install(testScheme)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tt := []struct {
+		name            string
+		existingObjects []runtime.Object
+		expectedDump    *testhelpers.GatherDump
+		expectedError   error
+	}{
+		{
+			name: "smoke test",
```

Comment:
should it include some extra cases? At least differentiate `--all-resources` from a basic run?

Reply:
Added test for all resources. I've tried a real external resource but fake client has a fixed scheme so I wired at least an unlikely internal resource stat is present in the scheme already.